### PR TITLE
Adds ApiMayChange and InternalApi - last round

### PR DESCRIPTION
**Pull request by renatocaval**
Last round for annotating public APIs with `@ApiMayChange` and internal APIs with `@InternalApi`

References #96

### CODE REVIEW

**Review by ignasi35 — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-projection-core/src/main/scala/akka/projection/MergeableOffset.scala [hunk 1]
```diff
@@ -4,4 +4,6 @@
 
 package akka.projection
 
+import akka.annotation.ApiMayChange
+@ApiMayChange
```
Comment (seglo) [inline akka-projection-core/src/main/scala/akka/projection/MergeableOffset.scala:8]: [minor] Line break? Not sure if this is something the formatter should have done, doesn't seem to matter.

File: akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala [hunk 2]
```diff
@@ -6,8 +6,9 @@ package akka.projection
 
 import scala.collection.immutable
 
+import akka.annotation.ApiMayChange
 import akka.util.ccompat.JavaConverters._
-
+@ApiMayChange
```
Comment (seglo) [inline akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala:11]: [minor] Line break

**Review by seglo — APPROVED** [review]
LGTM

File: akka-projection-core/src/main/scala/akka/projection/MergeableOffset.scala (hunk 1 above)
Comment (renatocaval) [inline akka-projection-core/src/main/scala/akka/projection/MergeableOffset.scala:8]:
```suggestion

@ApiMayChange
```

File: akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala (hunk 2 above)
Comment (renatocaval) [inline akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala:11]:
```suggestion

@ApiMayChange
```

**Pull request merged**